src/sources/file.rs

```diff
@@ -659,35 +723,70 @@ fn wrap_with_line_agg(
 }
 
 struct EventMetadata {
-    host_key: String,
+    host_key: Option<OwnedValuePath>,
     hostname: Option<String>,
-    file_key: Option<String>,
-    offset_key: Option<String>,
+    file_key: Option<OwnedValuePath>,
+    offset_key: Option<OwnedValuePath>,
 }
 
-fn create_event(line: Bytes, offset: u64, file: &str, meta: &EventMetadata) -> LogEvent {
+fn create_event(
+    line: Bytes,
+    offset: u64,
+    file: &str,
+    meta: &EventMetadata,
+    log_namespace: LogNamespace,
+) -> LogEvent {
     emit!(FileEventsReceived {
         count: 1,
         file,
         byte_size: line.len(),
     });
 
-    let mut event = LogEvent::from_bytes_legacy(&line);
+    let deserializer = BytesDeserializer::new();
```

Comment:
Might have missed this but can you create the deserializer once (elsewhere), and pass it into this function?

Reply:
Probably? I followed the same pattern that I had already merged here: https://github.com/vectordotdev/vector/blob/master/src/sources/aws_s3/sqs.rs#L519

Reply:
`BytesDeserializer` is very cheap to create, so I don't think this one matters too much

Reply:
cool, yeah just noticed it on this PR.